packaging/tags.py

```diff
@@ -700,18 +706,12 @@ def _manylinux_tags(linux, arch):
         for glibc_minor in range(glibc_max.minor, min_minor, -1):
             glibc_version = (glibc_max.major, glibc_minor)
             tag = "manylinux_{}_{}".format(*glibc_version)
-            # Once _is_manylinux_compatible() is True, it is True for any
-            # lower manylinux tag for this glibc major version.
-            is_compat = is_compat or _is_manylinux_compatible(tag, glibc_version)
-            if is_compat:
+            if _is_manylinux_compatible(tag, arch, glibc_version):
                 yield linux.replace("linux", tag)
-            # Handle the manylinux1, manylinux2010, manylinux2014 tags.
+            # Handle the legacy manylinux1, manylinux2010, manylinux2014 tags.
             if glibc_version in _LEGACY_MANYLINUX_MAP:
                 legacy_tag = _LEGACY_MANYLINUX_MAP[glibc_version]
-                is_compat = is_compat or _is_manylinux_compatible(
-                    legacy_tag, glibc_version
-                )
-                if is_compat:
+                if _is_manylinux_compatible(legacy_tag, arch, glibc_version):
                     yield linux.replace("linux", legacy_tag)
```

Comment:
We must check `_is_manylinux_compatible` each time, since the user-provided `_manylinux.manylinux_compatible` function is not required to follow any logic